### PR TITLE
Refactor GetItem tool

### PR DESCRIPTION
**Pull request by vkamra**
## Description

Refactors the GetItem tool to prep for adding a OneDrive cmd that can be used for micro-benchmarking

## Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :computer: CI/Deployment
- [ ] :hamster: Trivial/Minor

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/cmd/getM365/exchange/getItem.go [hunk 1]
```diff
@@ -24,52 +24,36 @@ import (
...
+	user = userFlag
+	tenant = tenantFlag
+
+	exCmd := &cobra.Command{
+		Use:   "exchange",
+		Short: "Get a M365ID item JSON",
+		RunE:  handleExchangeCmd,
+	}
+
+	fs := exCmd.PersistentFlags()
```
Comment (meain) [inline src/cmd/getM365/exchange/getItem.go:45]: I belive we are missing the user option here.
```
	fs.StringVar(&user, "user", "", "m365 user id of M365 user")
```

Comment (ryanfkeepers) [reply src/cmd/getM365/exchange/getItem.go:45]: They're introduced as persistent flags attached to the root command.  However, they're not going to be available to this package with this implementation..

File: src/cmd/getM365/exchange/getItem.go [hunk 2]
```diff
@@ -24,52 +24,36 @@ import (
...
 	fs.StringVar(&m365ID, "m365ID", "", "m365 identifier for object to be created")
 	fs.StringVar(&category, "category", "", "type of M365 data (contacts, email, events or files)") // files not supported
 
-	cobra.CheckErr(getCmd.MarkPersistentFlagRequired("user"))
-	cobra.CheckErr(getCmd.MarkPersistentFlagRequired("m365ID"))
-	cobra.CheckErr(getCmd.MarkPersistentFlagRequired("category"))
+	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("user"))
```
Comment (ryanfkeepers) [inline src/cmd/getM365/exchange/getItem.go:49]: The user is not declared for the `exCmd` flag set.  This need to be moved to main.

File: src/cmd/getM365/exchange/getItem.go [hunk 3]
```diff
@@ -24,52 +24,36 @@ import (
...
+	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("m365ID"))
+	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("category"))
```
Comment (ryanfkeepers) [inline src/cmd/getM365/exchange/getItem.go:51]: Not that it matters much, but unless we expect subcommands from here, these flags can be declared non-persistent.

File: src/cmd/getM365/exchange/getItem.go [hunk 4]
```diff
@@ -24,52 +24,36 @@ import (
...
-
-	defer logger.Flush(ctx)
-
-	fs := getCmd.PersistentFlags()
-	fs.StringVar(&user, "user", "", "m365 user id of M365 user")
-	fs.StringVar(&tenant, "tenant", "",
-		"m365 Tenant: m365 identifier for the tenant, not required if active in OS Environment")
+func AddCommands(parent *cobra.Command, userFlag, tenantFlag string) {
+	user = userFlag
+	tenant = tenantFlag
```
Comment (ryanfkeepers) [inline src/cmd/getM365/exchange/getItem.go:37]: Not sure if you're trying to link references, or copy values, but either way it won't work.  Per references, the flag pointers wont update the exchange package vars.  Per values, flag values are not populated until command execution.

If you want to distribute a common set of flag vars, they'll need to go into a third package.  For example, see: src/cli/options/options.go.

Comment (vkamra) [reply src/cmd/getM365/exchange/getItem.go:37]: 🤦 - I should really test this code before pushing.

**Comment by meain** [comment]
Closing in favor of https://github.com/alcionai/corso/pull/2791

**Pull request closed without merging**